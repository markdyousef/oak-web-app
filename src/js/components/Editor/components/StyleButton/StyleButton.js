import React, { Component, PropTypes } from 'react';
import ActionIcon from '../ActionIcon';

export default class StyleButton extends Component {
    static propTypes = {
        onToggle: PropTypes.func.isRequired,
        style: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        active: PropTypes.bool.isRequired,
        Icon: PropTypes.func
    }
    static defaultProps = {
        Icon: null
    }
    constructor() {
        super();
        this.state = {};
    }
    onToggle = (e) => {
        const { onToggle, style } = this.props;
        e.preventDefault();
        onToggle(style);
    }
    render() {
        const { label, active, Icon } = this.props;
        return (
            <ActionIcon active={active} onClick={this.onToggle}>
                {Icon && <Icon />}
                {label}
            </ActionIcon>
        );
    }
}
