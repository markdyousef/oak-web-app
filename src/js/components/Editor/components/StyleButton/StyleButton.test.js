import React from 'react';
import renderer from 'react-test-renderer';
import StyleButton from './StyleButton';

it('renders component correctly', () => {
    const props = {
        onToggle: () => {},
        style: '',
        label: '',
        active: false
    };
    const component = renderer.create(<StyleButton {...props} />);
    expect(component).toMatchSnapshot(component);
});
