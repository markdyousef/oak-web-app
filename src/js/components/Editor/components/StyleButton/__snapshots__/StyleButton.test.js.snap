exports[`[object Object] 1`] = `
<button
  className="jUAUJa"
  onClick={[Function]}>
  
</button>
`;
